csrc/scheduler/normalization_utils.cpp

```diff
@@ -722,5 +722,26 @@ getOptionalInnerOuterPersistentBufferBatches(
   }
 }
 
+TensorView* getReferenceReductionTv(
+    const std::vector<TensorView*>& reduction_tvs) {
+  TensorView* first_inner_tv = nullptr;
+  TensorView* first_outer_tv = nullptr;
+  for (auto tv : reduction_tvs) {
+    bool is_inner = scheduler_utils::isFastestDimReduction(tv);
+
+    if (is_inner && !first_inner_tv) {
+      first_inner_tv = tv;
+    } else if (!is_inner && !first_outer_tv) {
+      first_outer_tv = tv;
+    }
+
+    if (first_inner_tv && first_outer_tv) {
```

Comment:
Why outer reduction tensors matter?

Reply:
If there is an outer reduction and an inner reduction, it means we are doing combined reduction. It is used as an implicit flag indicating the reduction type.

Reply:
Wouldn't it be just much simpler to use this function only for the combined case? For the non-combined cases, we just need to pick the first one.

Reply:
It is a small utility function, so the current design allows this function being called without knowing the reduction type, same API for all reduction types. 
If change to only used by combined case, we need to alter some lines of code (in `getPersistentHeuristics`) to get the reduction type before calling this function. Is this prefered?

Reply:
I'm not sure what you mean by:

> to get the reduction type before calling this function. I

Since we are going to decompose the normalization scheduler, is this an issue?

In any case, as I said above, this is an example why we want to decompose the normalization scheduler. We can greatly simplify the code if this is only used for the combined case, and we can also have an appropriate error check. In the current version, if it's accidentally used for, say, just an outer normalization fusion, it would simply return an outer-reduction tensor, which makes no sense.

Reply:
If considering the decomposed version there is no issue. In this PR, I just move back to a version with minimum change to fix the bug. Two step: (1) check reduction type (2) return appropriate reference tv based on reduction type.